### PR TITLE
Toybox bump 0.7.5 -> 0.8.2

### PR DESCRIPTION
**Pull request by miartad**
Hi,

I was playing with yocto and toybox and as a result I have a few patches.

- allow defconfig in bbappend (similar to busybox or kernel)
- version bump and compile fix for new version

As for the CFLAGS and CC seems like there is different expectation from toybox as to what should
be provided, Im not sure my way of fixing is the most "clean" in yocto, but it seems to work fine.

I have tested compilation with/without defconfig on ARM. I have not tested running the executable on the target, but if anyone finds useful feel free to test or merge.

Best Regards,

Adam

### CODE REVIEW

File: meta-oe/recipes-core/toybox/toybox_0.8.2.bb [hunk 1]
```diff
@@ -3,21 +3,27 @@ HOMEPAGE = "http://www.landley.net/toybox/"
 DEPENDS = "attr virtual/crypt"
 
 LICENSE = "BSD-0-Clause"
-LIC_FILES_CHKSUM = "file://LICENSE;md5=f0b8b3dd6431bcaa245da0a08bd0d511"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=78659a599b9325da368f2f1eb88f19c7"
```
Comment (kraj) [inline meta-oe/recipes-core/toybox/toybox_0.8.2.bb:6]: what change in License is causing this checksum to change ? Please add that info to commit message

**Comment by miartad** [comment]
thanks, update the commit message detailing the license change

File: meta-oe/recipes-core/toybox/toybox_0.8.2.bb (hunk 1 above)
Comment (miartad) [reply meta-oe/recipes-core/toybox/toybox_0.8.2.bb:6]: updated commit message

**Comment by kraj** [comment]
fails on mips https://errors.yoctoproject.org/Errors/Details/411605/
fails with clang https://errors.yoctoproject.org/Errors/Details/411596/

**Comment by miartad** [comment]
> fails on mips https://errors.yoctoproject.org/Errors/Details/411605/

issue with a build script, similar issue that this request fixes but for some reason on mips it still misbehaves

> fails with clang https://errors.yoctoproject.org/Errors/Details/411596/

seem like this needs toybox fix

I'm having some issues running yoe builds, as I normally use poky without docker, but I will dig into this over the weekend

**Comment by kraj** [comment]
> > fails on mips https://errors.yoctoproject.org/Errors/Details/411605/
> 
> issue with a build script, similar to the issue this request fixes but for some reason on mips it still misbehaves
> 
> > fails with clang https://errors.yoctoproject.org/Errors/Details/411596/
> 
> seem like this needs toybox fix
> 
> I'm having some issues running yoe builds, as I normally use poky without docker, but I will dig into this over the weekend

you can disable docker 
```
cp local.sh.example local.sh
```
then in local.sh add 
`export DOCKER_REPO=none`

Then 
```
. ./qemumips-envsetup.sh 
bitbake toybox
```
This will build without docker.

**Comment by kraj** [comment]
parts of this pull are now merged please rebase on master and resend.

**Comment by miartad** [comment]
sorry got sidetracked, I'm still thinking about this + I will be working on basic toybox-inittab package

**Comment by kraj** [comment]
squashed/edited/merged eaf4a81244

**Pull request closed without merging**